### PR TITLE
Swift 4.2 migration 🚀

### PR DESCRIPTION
**Pull request by IvoPintodaSilva**
Migrating the codebase to Swift 4.2

### CODE REVIEW

File: Alicerce.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1467,21 +1449,24 @@
 				TargetAttributes = {
 					0A266F801ED59DC7009CD0D7 = {
 						CreatedOnToolsVersion = 8.3.2;
-						LastSwiftMigration = 0910;
-						ProvisioningStyle = Manual;
+						DevelopmentTeam = 74L475LDQT;
```
Comment (beloso) [inline Alicerce.xcodeproj/project.pbxproj]: I believe we should not be committing this.

File: Alicerce.podspec [hunk 2]
```diff
@@ -26,6 +26,4 @@ Pod::Spec.new do |s|
 
     s.frameworks    = [ 'Foundation', 'UIKit', 'CoreData', 'Security' ]
     s.dependency 'Result', '~> 4.0'
-
-    s.script_phase  = { :name => 'CommonCrypto', :script => '${PODS_TARGET_SRCROOT}/install_common_crypto.sh', :execution_position => :before_compile }
```
Comment (p4checo) [inline Alicerce.podspec]: We should also remove the `s.preserve_paths` line, and delete the install_common_crypto.sh file 🗑

File: README.md [hunk 3]
```diff
@@ -48,7 +48,7 @@ TODO
 
 ## Installation 🔧
 
-Alicerce supports iOS 9.0+ and requires Xcode 9.3+.
+Alicerce supports iOS 9.0+ and requires Xcode 10+.
```
Comment (p4checo) [inline README.md]: We should probably add a note that for Xcode 9.3 compatibility (Swift 4.0), Alicerce 0.3.0 should be used 💅

**Comment by IvoPintodaSilva** [comment]
Updated with your suggestions @beloso and @p4checo 👍

File: Sources/UIKit/UIViewController.swift [hunk 4]
```diff
@@ -32,8 +32,8 @@ public extension UIViewController {
     ///   - unselectedRenderMode: UIImageRenderingMode to the unselected image. Default .alwaysOriginal
```
Comment (filipe-lemos) [inline Sources/UIKit/UIViewController.swift]: Documentation is still referencing old types.

Comment (IvoPintodaSilva) [reply Sources/UIKit/UIViewController.swift]: Nice catch! 👍

**Pull request merged**